packages/zkdb/src/sdk/global/zkdatabase-client.ts

```diff
@@ -7,28 +7,61 @@ import { GlobalContext } from '../interfaces/global-context.js';
 import { ZKDatabaseImpl } from '../impl/database.js';
 import { MinaBlockchainImpl } from '../impl/blokchain.js';
 import { GlobalContextImpl } from '../impl/global-context.js';
+import { ApiClient, IApiClient } from '@zkdb/api';
+import storage from '../../storage/storage.js';
+
 class ZKDatabaseClient {
+  private apiClient: IApiClient<any>;
+
+  public connect(url: string, signer: Signer) {
+    this.initializeApiClient(url);
+    this.setSigner(signer);
+  }
+
+  private initializeApiClient(url: string) {
+    this.apiClient = ApiClient.newInstance(url);
+    this.apiClient.api.setContext(() => {
+      return storage.getAccessToken();
+    });
+  }
 
   public get auth(): Authenticator {
-    return new Authenticator(getSigner());
+    if (this.apiClient) {
+      return new Authenticator(getSigner(), this.apiClient);
```

Comment:
`getSigner()` should be implemented in `async newInstance()` function instead of independent function.